src/libs/ajax.js

```diff
@@ -1130,6 +1130,21 @@ const AzureStorage = signal => ({
         storageContainerName: container.resourceAttributes.azureStorageContainer.storageContainerName
       }
     }
+  },
+
+  resources: async workspaceId => {
+    const res = await fetchWorkspaceManager(`workspaces/v1/${workspaceId}/resources`, _.merge(authOpts(), { signal }))
+    return res.json()
+  },
+
+  sasToken: async workspaceId => {
+    const resourcesResponse = await AzureStorage(signal).resources(workspaceId)
+    const container = _.find(resource => resource.metadata.resourceType === 'AZURE_STORAGE_CONTAINER', resourcesResponse.resources)
+
+    const tokenResponse = await fetchWorkspaceManager(`workspaces/v1/${workspaceId}/resources/controlled/azure/storageContainer/${container.metadata.resourceId}/getSasToken`,
```

Comment:
Do you want to guard against the storage container not existing (which is the case for old workspaces)? I chose to do it in my code (line 1115), but also recognize that the additional complexity may not be worth carrying forward.

Note also that I am looking up the storage account first (assuming there will only be 1), and then finding the storage container that matches the storage account. You are assuming there is only one storage container… which is probably fine. But I am starting to think that perhaps just fold the SasToken into my `details` method so that we aren't doing this storage container lookup twice (and assured it is in fact the same storage container). The the details method would return {location, storageContainerName, sasToken}. What do you think?

Reply:
Yeah I think that makes sense so both code paths are in agreement about the container we're looking at.